### ISSUE TITLE
Modules: `export from` jumps load ordering

### ISSUE THREAD

**Issue by twooster** [meteor/meteor#9504]
**Meteor Version: 1.6**
**OS: Ubuntu**

Running in non-production, if that matters.

**Expected Behavior:**
Given the following files:
`index.js`:
```
import './a.js';
export { b } from './b.js'
```

`a.js`
```
console.log('a');
```

`b.js`:
```
console.log('b');
export const b = 1;
```

I would expect to see the following output:

```
a
b
```

**Actual Behavior:**

Instead I see:

```
b
a
```

However, if `index.js` instead reads:
```
import './a.js';
import { b } from './b.js';
export { b };
```

The output is, indeed, correct.

**Extra Notes:**
Looking at the generated output, I see, basically this:

Broken case:
```
module.watch(require("./b.js"), {
  b: function (v) {
    exports.b = v;
  }
}, 0);
module.watch(require("./a.js"));
```

Functional case:
```
module.export({
  b: function () {
    return b;
  }
});
module.watch(require("./a.js"));
var b = void 0;
module.watch(require("./b.js"), {
  b: function (v) {
    b = v;
  }
}, 0);
```

Y'all know your own codebase, but I suspect that since exports are clearly defined first in the generated output, someone tried to optimize a bit too much collapsing `export from` into a single statement, causing the import load ordering to be jumped.

**Comment by hwillson** [comment]
Confirmed - thanks @twooster, we'll take a look.

**Comment by benjamn** [comment]
This should be fixed if you run `meteor update --release 1.6.1-rc.2`!
